products/extensions/b2b-suite-migration/development/adding-component.md

```diff
@@ -0,0 +1,392 @@
+- Optionally define `baseConditions()`, `baseDefaultValues()`, and `baseFallbackValues()`.
+- Tag the class with `b2b.migration.configurator`.
+
+   **Example**:
+
+   ```xml
+   <service id="Shopware\Commercial\B2B\B2BSuiteMigration\Components\EmployeeManagement\EmployeeManagementMigrationConfigurator">
+      <tag name="b2b.migration.configurator" priority="9000"/>
+   </service>
+   ```
+
+  ```php
+  class EmployeeManagementMigrationConfigurator extends AbstractB2BMigrationConfigurator
+  {
+    public function getName(): string
+    {
+        return 'employee_management';
+    }
+
+    public function configPath(): string
+    {
+        return 'path/to/your/xml/mapping/file.xml';
+    }
+  
+    protected function baseMigrationProcesses(): array
+    {
+        return [
+            // should match the <name> in the XML mapping file
+            'migration_b2b_component_business_partner',
+        ];
+    }
+    ...
+  }
+  ```
+
+## Create XML Mapping File
+
+### Entity Definition
+
+   ```xml
+   <migration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../../../Core/Resources/Schema/Xml/migration-1.0.xsd">
+    <entity>
+      <name>migration_b2b_component_business_partner</name>
+      <source>b2b_customer_data</source>
+      <source_primary_key>customer_id</source_primary_key>
+      <target>b2b_business_partner</target>
+      <target_primary_key>id</target_primary_key>
+      <fields>
+        ...
+      </fields>
+    </entity>
+  </migration>
+   ```
+
+- **name**: Unique migration process identifier.
+- **source**: Source table name.
+- **source_primary_key**: Source table primary key (default: `id`).
+- **target**: Target table name.
+- **target_primary_key**: Target table primary key (default: `id`).
+
+### Fields
+
+```xml
+<fields>
+  <field source="customer_id" target="customer_id"/>
+  <field source="created_at" target="created_at"/>
+</fields>
+```
+
+- **fields**: Contains field mappings between source and target tables.
+- **field**: Defines how to map fields from the source to the target.
+
+## Field Mapping Configuration
+
+Field mappings define how data is transferred from source tables in B2B Suite to target tables in B2B Commercial. These mappings are specified in the XML configuration within the `<fields>` element. This section explains the different mapping types and their syntax.
+
+### One-to-One Mapping
+
+A one-to-one mapping directly maps a field from the source table to a field in the target table.
+
+**Example**:
+
+```xml
+<field source="customer_id" target="customer_id"/>
+```
+
+- **source**: The field name in the source table (e.g., `customer_id` in `b2b_customer_data`).
+- **target**: The field name in the target table (e.g., `customer_id` in `b2b_business_partner`).
+
+::: info
+Use one-to-one mappings for straightforward field transfers where no transformation is needed.
+:::  
+
+### Relational Mapping
```

Comment:
How are you dealing with mapping foreign keys? e.g. the employee gets a new UUID during migration, all other migrated entities like quotes need to reference this new UUID instead of the one from the source

Or do you always just reuse the UUID from the source data, I guess the primary keys in the source data are also all UUIDs so this is a possibility?